File: C++/OOP/Encapsulation/Encapsulation (example 2).cpp
#include <iostream>
using namespace std;

class Student{
    private:
        string name;
        int id;

    public:
        void setInfo(string studentName, int studentID){
            name = studentName;
            id = studentID;
        }

        void getInfo(){
            cout << "Name: " << name << endl;
            cout << "ID  : " << id << endl;
        }
};

int main(){
    Student naiem;
    naiem.setInfo("Abdullah Al Naiem", 123456);
    naiem.getInfo();

    cout << endl;

    Student sakib;
    sakib.setInfo("Sakibur Rahman", 456123);
    sakib.getInfo();
}

File: Python/OOP/Encapsulation/Encapsulation (example 2).py
# Python does not have the private keyword, unlike some other object oriented languages.
# But encapsulation can be done with the following method (member can be private by double underscore)
class Student:
    def setInfo(self, studentName, studentID):
        self.__name = studentName
        self.__id = studentID

    def getInfo(self):
        print("Name: ", self.__name)
        print("ID  : ", self.__id)


class encap:
    naiem = Student()
    naiem.setInfo("Abdullah Al Naiem", 123456)
    naiem.getInfo()

    print()

    sakib = Student()
    sakib.setInfo("Sakibur Rahman", 456123)
    sakib.getInfo()